cosmos/models/Workflow.py

```diff
@@ -560,10 +562,21 @@ def _process_finished_tasks(jobmanager):
     for task in jobmanager.get_finished_tasks():
         if task.NOOP or task.exit_status == 0:
             task.status = TaskStatus.successful
-            yield task
+        elif task.scheduling_error:
```

Comment:
That's a very good question! unfortunately I don't have a very good answer. fatigue? :-)